src/platform/intel/cavs/platform.c

```diff
@@ -270,6 +270,10 @@ static void platform_init_hw(void)
 		LPGPDMA_CHOSEL_FLAG | LPGPDMA_CTLOSEL_FLAG);
 	io_reg_write(DSP_INIT_LPGPDMA(1),
 		LPGPDMA_CHOSEL_FLAG | LPGPDMA_CTLOSEL_FLAG);
+#if defined(CONFIG_DMIC) && defined(CONFIG_ICELAKE)
```

Comment:
We need this also for cannonlake (and suecreek?). It would be better to have some better defines, because it's for cavs1.8+ and listing all these platforms in places like that will hurt us, but for now you can just go for `!defined(CONFIG_APOLLOLAKE)` because other cavs platforms need this.

Reply:
Yeah, I think @singalsu had some definitions for DMIC IP versions ?